### ISSUE TITLE
"structure" keyword isn't respected. (1.16.5)

### ISSUE THREAD

**Issue by jelaw21** [McJty/InControl#255]
Spawner.json
```
{
	  "mob":"aoa3:hauling_master",
	  "persecond": 1,
		"attempts": 20,
		
		"amount": {
		  "minimum": 1,
		  "maximum": 20
		},
	  "conditions":{
		  "norestrictions":true,
		  "dimension":"minecraft:overworld"
	  }
  }
```
Spawn.json
```
{
		"mob":"aoa3:hauling_master",
		"dimension":"minecraft:overworld",
		"result":"allow"		
	}
```
Produces:
![Inked2021-04-24_14 57 28_LI](https://user-images.githubusercontent.com/21100827/115970076-327b5200-a50e-11eb-8411-8765b01a56d6.jpg)

Same Spawner file
Spawn.json
```
{
		"mob":"aoa3:hauling_master",
		"structure":"Village",
		"dimension":"minecraft:overworld",
		"result":"allow"		
	},
	{
		"mob":"aoa3:hauling_master",
		"dimension":"minecraft:overworld",
		"result":"deny"
	}
```
Produces:
![2021-04-24_14 58 11](https://user-images.githubusercontent.com/21100827/115970109-6b1b2b80-a50e-11eb-9036-d70ec59a55ab.png)

There are no aoa3:hauling_master spawning in the village, even with the structures keyword.

**Comment by McJty** [comment]
'Village' is not a valid structure. Must be something like 'minecraft:village'. At least a proper ID

**Comment by jelaw21** [comment]
Ok. Will test tomorrow.

**Comment by JoelZBub** [comment]
I've tested this with "minecraft:cow"  and "minecraft:village" and this is a valid issue.
```
[                      	
    {                  	
        "onjoin": true,	
        "structure": "minecraft:village",
        "mob": "minecraft:cow",
        "result": "allow"        
    },
    {
        "dimension":"minecraft:overworld",
        "mob":"minecraft:cow",
        "result": "deny"
    }	
]
```
The above produces no cow spawns. 
Removing the "onjoin" condition from the allow rule allows cows to spawn in the overworld in various places. 
Adding "onjoin": true to just the deny rule prevents all cow spawns.
I also tested the above adding "minecraft:overworld" to the allow rule, and even added a spawner rule to add cows to the overworld.

**Comment by McJty** [comment]
Should be fixed in the upcoming 1.16-5.1.7
